pylint/extensions/magic_value.py

```diff
@@ -0,0 +1,91 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+"""Checks for magic numbers instead of literals."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from astroid import nodes
+
+from pylint.checkers import BaseChecker, utils
+from pylint.checkers.base.comparison_checker import ComparisonChecker
+from pylint.interfaces import HIGH
+
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
+
+
+class MagicValueChecker(BaseChecker):
```

Comment:
```suggestion
class MagicValueChecker(BaseChecker, _ArgumentsProvider):
```

I thought this was required to auto register options. But it's not working and I had to read some code to find about that. @DanielNoord do you know how to register the options ? I'm going to document this in the contributor guide because I don't think there's anything yet.

Reply:
`BaseChecker` inherits from `_ArgumentsProvider` itself.

@orSolocate  I believe the issue is because of the use of `"` in the `pylintrc` file. Strings don't need to be quoted.

Reply:
I expected to see the new option in the ``--help`` is it supposed to be printed ?

Reply:
Yes, isn't it?

Reply:
Let me recheck that :)

Reply:
My bad we need to load the plugin first of course.
```
(venv)  pierre@pylint  ~/pylint   orsolocate/magic_numbers  pylint --help|grep magic                                             
(venv)  ✘ pierre@pylint  ~/pylint   orsolocate/magic_numbers  
pylint --load-plugins=pylint.extensions.magic_value --help|grep magic
  --valid-magic-values <method names>
                        List of valid magic numbers that `magic-value-compare`
```

Reply:
I tried to remove the `""`, still not working.. also the docs test is failing and I don't see why, there was an issue with not adding `.rc` file in `good/bad.py`  but  I thought that was fixed.

Reply:
> `BaseChecker` inherits from `_ArgumentsProvider` itself.
> 
> @orSolocate I believe the issue is because of the use of `"` in the `pylintrc` file. Strings don't need to be quoted.

How do you do an empty string then? maybe there was an error there?
`valid-magic-values=-1, 0, 1,, __main__`

Reply:
I'm not sure `ini` files support empty strings. I'll have a look at this somewhere this week. Probably around the weekend, bit busy currently.

Reply:
I ran `tox -e docs` like in the good old days and found that the documentation issue was because there was a missing `.` in the end of line in checker description..

The `primer` test fails because when run it shows this error, look like parsing error:
`pylint: error: argument --valid-magic-values: expected one argument`

@DanielNoord  - Could you help me with it please?

P.S. why did we remove the `tox -e docs` to check docs locally from contribution page? Maybe it is a good idea to put it back..

Reply:
I don't think we removed tox -e docs ?

Reply:
The primer issue is because of the default. By using `-1,0,...` there is a `-` at the beginning of the value and `argparse` chokes on this. Changing it around to have the `-1` at the second index fixes the issues and parses correctly.

Reply:
> I don't think we removed tox -e docs ?

From [here ](https://pylint.pycqa.org/en/latest/development_guide/contributor_guide/tests/launching_test.html)i meant.

That is were it used to be.

Anyway I would assume it would be under the [testing chapter](https://pylint.pycqa.org/en/latest/development_guide/contributor_guide/tests/index.html) but I can't find it.. If it exist somewhere else it should be there as well I think..

Reply:
Right, even if tox -e docs still exits for the CI, in the doc we documented something faster: https://pylint.pycqa.org/en/latest/development_guide/contributor_guide/contribute.html#building-the-documentation

Reply:
> The primer issue is because of the default. By using `-1,0,...` there is a `-` at the beginning of the value and `argparse` chokes on this. Changing it around to have the `-1` at the second index fixes the issues and parses correctly.

Thanks for checking it for me, the `primer` tests pass now.
The `tests` section fails though, seems like for parsing error of the same line, this is the error message I saw [in CI](https://github.com/PyCQA/pylint/actions/runs/3218362280/jobs/5262443518):

```
usage: pylint-config [options]
pylint-config: error: argument config_subcommand: invalid choice: '0,-1,1,__main__' (choose from 'generate')
```

@DanielNoord  - Could you please help me out?

Reply:
Tests pass if you remove the value from the `pylintrc` file. I'm not sure why and the error isn't really elegant but I can explain some part of it: your adding configuration for a plugin that is not being loaded in that `pylintrc` so `pylint` doesn't know what to do with those values.
It's very weird to throws this particular error and we should probably fix it, but removing would fix this.

Reply:
That makes sense actually, because it is an extension checker, it is not registers by default including the `options` section of it.
I will add to my default also `[]`, `{}` and `""`, because otherwise using this checker is even more annoying. 

Do you want me to open a bug for unclear message with pylintrc parsing error?

Reply:
Yeah could you open an issue for it? This is really ugly.